mypy/checker.py

```diff
@@ -3078,6 +3078,11 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
             with self.enter_final_context(s.is_final_def):
                 self.check_assignment(s.lvalues[-1], s.rvalue, s.type is None, s.new_syntax)
 
+        for lv in s.lvalues:
+            if isinstance(lv, MemberExpr):
+                if lv.name == "__class__":
+                    self.fail("Assignment to '__class__' is unsafe and not allowed", lv)
```

Comment:
This wording is too strong - assignments are fine themselves, there's nothing inherently unsafe with doing this, it's not a security vuln. Probably something along the lines of

```suggestion
                    self.fail("Assignment to '__class__' is not supported", lv)
```

(probably accompanied by a `self.note` explaining it further) would be less intrusive.